### PR TITLE
GT-389 Async Client

### PR DESCRIPTION
**Pull request by jwierzbo**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //
 // DISCLAIMER
 //
-// Copyright 2017 ArangoDB GmbH, Cologne, Germany
+// Copyright 2013 ArangoDB GmbH, Cologne, Germany
```
Comment (informalict) [inline client.go:4]: 2013 -> 2023

File: context.go [hunk 2]
```diff
@@ -561,3 +575,19 @@ func withDocumentAt(ctx context.Context, index int) (context.Context, error) {
 
 	return ctx, nil
 }
+
+//
+// READ METHODS
+//
+
+func IsAsyncIDSet(ctx context.Context) (string, bool) {
```
Comment (informalict) [inline context.go:583]: comment?

File: asyncjob.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+type AsyncJobDeleteType string
+
+const (
+	DeleteAllJobs     AsyncJobDeleteType = "all"
+	DeleteExpiredJobs AsyncJobDeleteType = "expired"
+	DeleteSingleJob   AsyncJobDeleteType = "single"
+)
+
+type AsyncJobDeleteOptions struct {
+	JobID string    `json:"id,omitempty"`
```
Comment (informalict) [inline asyncjob.go:61]: missing fields comments

File: asyncjob_impl.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+		return false, WithStack(err)
+	}
+	return data.Result, nil
+}
+
+type deleteResponse struct {
+	Result bool `json:"result"`
+}
+
+func (c *clientAsyncJob) Delete(ctx context.Context, deleteType AsyncJobDeleteType, opts *AsyncJobDeleteOptions) (bool, error) {
```
Comment (informalict) [inline asyncjob_impl.go:122]: missing functions comments? this function and all above functions.

**Review by nikita-vanyasin — COMMENTED** [review]
Maybe we need that for v2 also?

File: asyncjob_impl.go (hunk 4 above)
Comment (jwierzbo) [reply asyncjob_impl.go:122]: as we agreed we do not comment implementation methods

**Comment by jwierzbo** [comment]
> Maybe we need that for v2 also?

In separate PR, since there will be more effort required (e.g. passing custom header is not implemented yet in V2)

**Comment by jwierzbo** [comment]
will be continued in https://github.com/arangodb/go-driver/pull/525

**Pull request closed without merging**